src/recon_buildblock/PoissonLogLikelihoodWithLinearModelForMeanAndListModeDataWithProjMatrixByBin.cxx

```diff
@@ -150,38 +266,146 @@ PoissonLogLikelihoodWithLinearModelForMeanAndListModeDataWithProjMatrixByBin<Tar
         scanner_sptr->get_num_rings()-1;
     }
 
-     
-  if (this->additive_projection_data_filename != "0") 
-    { 
+  if (this->additive_projection_data_filename != "0")
+    {
       info(boost::format("Reading additive projdata data '%1%'")
            % additive_projection_data_filename  );
-      shared_ptr <ProjData> temp_additive_proj_data_sptr =  
-        ProjData::read_from_file(this->additive_projection_data_filename); 
+      shared_ptr <ProjData> temp_additive_proj_data_sptr =
+        ProjData::read_from_file(this->additive_projection_data_filename);
       this->additive_proj_data_sptr.reset(new ProjDataInMemory(* temp_additive_proj_data_sptr));
-    } 
-  
-
-  this->proj_data_info_cyl_uncompressed_ptr.reset(
-    dynamic_cast<ProjDataInfoCylindricalNoArcCorr *>(
-    ProjDataInfo::ProjDataInfoCTI(scanner_sptr, 
-                  1, this->max_ring_difference_num_to_process,
-                  scanner_sptr->get_num_detectors_per_ring()/2,
-                  scanner_sptr->get_default_num_arccorrected_bins(), 
-				  false)));
+    }
+
+   proj_data_info_cyl_sptr = this->list_mode_data_sptr->get_proj_data_info_sptr()->create_shared_clone();
+
+   if (max_ring_difference_num_to_process > proj_data_info_cyl_sptr->get_max_segment_num())
+   {
+       warning("In the parameter file, the 'maximum ring difference' is larger than the number of segments"
+               "in the emission header. Abort.");
```

Comment:
I guess this would have to be "listmode file", not "emission header" (although I'm not sure that really want this functionality)